### PR TITLE
Get voting power

### PR DESCRIPTION
**Pull request by abhishek-k-sharma-lab49**
Governor voting with linear and quadratic calculations complete.

### CODE REVIEW

File: contracts/governance/GODHolder.sol [hunk 1]
```diff
@@ -77,11 +77,11 @@ contract GODHolder is IGODHolder, Initializable {
     function grabTokensFromUser(
         address user
     ) external override returns (uint256 amount) {
-        if (godTokenForUsers[user] > 0) {
+        if (godTokenForUsers[user] > 0 && _token.balanceOf(user) == 0) {
```
Comment (shivamgarglab49) [inline contracts/governance/GODHolder.sol:80]: we can store _token.balanceOf(user) result in variable for further use in next line.

Also we can use short hand operator for adding amount like -> godTokenForUsers[user] += amount;

File: contracts/governance/GovernorCountingSimpleInternal.sol [hunk 2]
```diff
@@ -266,7 +272,7 @@ abstract contract GovernorCountingSimpleInternal is
     function quorum(
         uint256
     ) public view override(IGovernorUpgradeable) returns (uint256) {
-        uint256 totalSupply = token.totalSupply();
+        uint256 totalSupply =  isLinearVoting ? token.totalSupply() : _sqrt(token.totalSupply());
```
Comment (amandeeprehal-lab49) [inline contracts/governance/GovernorCountingSimpleInternal.sol:275]: Pls don't call `token.totalSupply()` mulitple times. Just call its once store result locally as it is remote function so not efficient

Comment (amandeeprehal-lab49) [inline contracts/governance/GovernorCountingSimpleInternal.sol:275]: Why we need sqrt here?

File: deployment/state/contracts.json [hunk 3]
```diff
@@ -158,5 +158,50 @@
         "hash": "555a9b88e50a37737b7f946c3c5e5a23",
         "transparentProxyAddress": "0x0000000000000000000000000000000000002321",
         "transparentProxyId": "0.0.8993"
+    },
+    {
+        "name": "factory",
```
Comment (amandeeprehal-lab49) [inline deployment/state/contracts.json:163]: Remove this factory pls

File: deployment/state/contracts.json [hunk 4]
```diff
@@ -158,5 +158,50 @@
...
+        "name": "governortokencreate",
+        "id": "0.0.3121070",
+        "address": "0x00000000000000000000000000000000002f9fae",
+        "timestamp": "2023-02-01T06:41:51.986Z",
+        "hash": "180dc4c5342cedf6b9dc57247dd75253",
+        "transparentProxyAddress": "0x00000000000000000000000000000000002f9fb1",
+        "transparentProxyId": "0.0.3121073"
+    },
+    {
+        "name": "governortokencreate",
```
Comment (amandeeprehal-lab49) [inline deployment/state/contracts.json:199]: Keep latest and remove others.

File: e2e-test/step-definitions/GovernorTest.ts [hunk 5]
```diff
@@ -90,6 +90,7 @@ export class GovernorSteps {
 
   @then(/user verify that proposal state is (\d*)/, undefined, 30000)
   public async verifyProposaState(proposalState: string): Promise<void> {
+    console.log(`ProposalId: ${proposalID}`);
```
Comment (amandeeprehal-lab49) [inline e2e-test/step-definitions/GovernorTest.ts:93]: Why you need this?

File: test/governer-test.ts [hunk 6]
```diff
@@ -245,6 +254,11 @@ describe("Governor Tests", function () {
 
       await instance.castVote(proposalId, 1);
       const voteSucceeded1 = await instance.voteSucceeded(proposalId);
+      await verifyProposalVotes(instance, proposalId, {
+        abstainVotes: 0,
+        againstVotes: 0,
+        forVotes: 19 * precision,
```
Comment (amandeeprehal-lab49) [inline test/governer-test.ts:260]: Pls use context `twentyPercent` - 1.

File: e2e-test/business/Governor.ts [hunk 7]
```diff
@@ -33,7 +33,8 @@ export default class Governor {
       .addUint256(votingPeriod)
       .addAddress(htsServiceAddress)
       .addAddress(godHolderProxyAdd)
-      .addUint256(defaultQuorumThresholdValue);
+      .addUint256(defaultQuorumThresholdValue)
+      .addBool(true);
```
Comment (amandeeprehal-lab49) [inline e2e-test/business/Governor.ts:37]: why this?

File: test/governer-test.ts [hunk 8]
```diff
@@ -62,6 +62,12 @@ describe("Governor Tests", function () {
     return basicDeployments(mockBaseHTS);
   }
 
+  async function deployFixtureForQuadratic() {
```
Comment (amandeeprehal-lab49) [inline test/governer-test.ts:65]: why this?

**Review by amandeeprehal-lab49 — APPROVED** [review]

**Pull request merged**